query_select.go

```diff
@@ -979,20 +979,25 @@ func (q *SelectQuery) scanAndCountConcurrently(
 	var mu sync.Mutex
 	var firstErr error
 
+	// FIXME: clone should not be needed, because the query is not modified here
+	// and should not be implicitly modified by the Bun lib.
 	countQuery := q.Clone()
 
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-
-		if err := q.Scan(ctx, dest...); err != nil {
-			mu.Lock()
-			if firstErr == nil {
-				firstErr = err
+	// Don't scan results if the user explicitly set Limit(-1).
+	if q.limit >= 0 {
```

Comment:
In ScanAndCount: when offset == 0 && limit == 0, 
it performs only a scan query and returns both the results and the count.

Now there’s a request to add logic: if offset < 0, it should perform only the count operation.
Wouldn’t it be better to move this logic up into ScanAndCount?

Also, could you briefly explain why limit might be less than 0? I imagine not everyone is familiar with that.